x/reports/internal/keeper/keeper.go

```diff
@@ -0,0 +1,100 @@
+package keeper
+
+import (
+	"bytes"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/desmos-labs/desmos/x/posts"
+	"github.com/desmos-labs/desmos/x/reports/internal/types"
+	"github.com/desmos-labs/desmos/x/reports/internal/types/models"
+)
+
+// Keeper maintains the link to data storage and exposes getter/setter methods for the various parts of the state machine
+type Keeper struct {
+	PostKeeper posts.Keeper // post's keeper to perform checks on the postIDs
```

Comment:
I've looked at the code and it seems to me that this isn't used anywhere. Am I right?

Reply:
it is used inside the `handler` to check if there is any post related to the ID given, otherwise anyone could report non-existent posts.

Reply:
@bragaz Can we encapsulate this operation inside a keeper method such as `CheckPostExistance` and make the `PostKeeper` private (renaming it `postsKeeper`)? This would hide this dependency that I think will result in a more easy to comprehend code

Reply:
we could have some problems with sim tests but I can try